### PR TITLE
Make some tests that should fail, actually fail.

### PR DESCRIPTION
**Pull request by cedvdb**
Make some tests that should fail, actually fail. The fact that the tests were engineered to succeed is concerning.

From those failing tests we can see that `orderBy` is failing, `endAt` and `startAt`

Side effect of closing https://github.com/atn832/fake_cloud_firestore/issues/204

### CODE REVIEW

File: test/mock_query_test.dart [hunk 1]
```diff
@@ -1232,10 +1359,13 @@ void main() {
...
       },
+      {
+        'name': 'Springfield',
+        'state': 'Missouri',
+      },
     ]);
 
-    snapshots = await baseQuery.endAt(['Springfield', 'Missouri']).get();
-
+    snapshots = await baseQuery.endAt(['Wellington']).get();
```
Comment (atn832) [inline test/mock_query_test.dart:1368]: Should it return everything because Wellington is alphabetically grater than Washington, or because "Wellington" as a value does not exist, and so it returns everything?

File: test/mock_query_test.dart [hunk 2]
```diff
@@ -1186,6 +1256,10 @@ void main() {
         'name': 'Springfield',
         'state': 'Wisconsin',
       },
+      {
+        'name': 'Washington',
```
Comment (atn832) [inline test/mock_query_test.dart:1260]: Nice catch, I wonder why we missed this.

File: test/mock_query_test.dart [hunk 3]
```diff
@@ -1150,12 +1150,79 @@ void main() {
...
+    await instance.collection('cities').doc().set({
+      'name': 'Washington',
+      'state': 'Washington',
+    });
+
     final baseQuery =
         instance.collection('cities').orderBy('name').orderBy('state');
 
-    var snapshots = await baseQuery.startAt(['Springfield']).get();
+    var snapshots = await baseQuery.startAt(['Los Angeles']).get();
```
Comment (atn832) [inline test/mock_query_test.dart:1219]: What made you want to change the value here? Now we won't know if the test passes because startAt returned everything, or because it really matched with the value.

File: test/mock_query_test.dart (hunk 1 above)
Comment (atn832) [reply test/mock_query_test.dart:1368]: If it's the latter, it'd be nice to add a comment to explain it, and perhaps change Wellington to some dummy value so that it is obvious why the query returns those results.

File: lib/src/mock_query.dart [hunk 4]
```diff
@@ -115,7 +115,7 @@ class MockQuery<T extends Object?> extends FakeQueryWithParent<T> {
       values: values,
       f: (docs, index, exactMatch) => docs.sublist(
             0,
-            index == -1 ? docs.length : index + 1,
```
Comment (atn832) [inline lib/src/mock_query.dart:118]: I don't know what the real Firestore behavior is, but in our previous implementation, if the endAt value wasn't found, we would return everything. With your change, now we will return empty. Is that what you are trying to achieve?

File: test/mock_query_test.dart (hunk 3 above)
Comment (cedvdb) [reply test/mock_query_test.dart:1219]: I think I did it to be sure it returned everything when the index was 0. I don't know exactly. I will revert this.

File: lib/src/mock_query.dart (hunk 4 above)
Comment (cedvdb) [reply lib/src/mock_query.dart:118]: I will revert this, although the next PR removes this

File: test/mock_query_test.dart (hunk 1 above)
Comment (cedvdb) [reply test/mock_query_test.dart:1368]: Wellington is alphabetically greater so it returns everything that is under it. Do you want me to rename wellington to something clearer ? Do you have a variable name in mind ?

Comment (atn832) [reply test/mock_query_test.dart:1368]: No, Wellington is fine. I just wasn't sure what the expected behavior was. Since edge cases with queries can get complicated, perhaps you can add a comment to explain that if the value is greater than all others, we expect everything to be returned. Thank you!

File: lib/src/mock_query.dart (hunk 4 above)
Comment (cedvdb) [reply lib/src/mock_query.dart:118]: Is it okay if I don't revert this as the next pull request essentially remove this ?  I'm afraid to have to rebase everything after this..

**Comment by cedvdb** [comment]
I've added comments but there is still one of your comment I did not address. Please let me know if my response sounds okay so we can move forward with the next pull requests.

File: lib/src/mock_query.dart (hunk 4 above)
Comment (atn832) [reply lib/src/mock_query.dart:118]: Sure, let's keep it simple for you.
